sway-core/src/decl_engine/associated_item_decl_id.rs

```diff
@@ -14,6 +16,17 @@ pub enum AssociatedItemDeclId {
     Type(DeclId<ty::TyTraitType>),
 }
 
+impl AssociatedItemDeclId {
+    pub fn span(&self, engines: &Engines) -> Span {
+        match self {
+            Self::TraitFn(decl_id) => engines.de().get(decl_id).span(),
+            Self::Function(decl_id) => engines.de().get(decl_id).span(),
+            Self::Constant(decl_id) => engines.de().get(decl_id).span(),
+            Self::Type(decl_id) => engines.de().get(decl_id).span(),
+        }
```

Comment:
We should probably split the match into a (private) `decl_id` function and make this a oneliner.

Reply:
Ah yes, never mind.